Assets/TrackingAI.cs

```diff
@@ -5,130 +5,136 @@
-		if (target != null)
-		{
-			int sign = Vector3.Cross (target.transform.position - this.transform.position, this.transform.up).z < 0 ? 1 : -1;
-			float angle = Vector3.Angle (target.transform.position - this.transform.position, this.transform.up);
-			float rate = degreesPerSecond * Time.deltaTime;
-
-			// if remaining angle is less than turn rate, use remaining angle (won't overshoot)
-			rate = angle < rate ? angle : rate;
-
-			transform.Rotate (Vector3.forward, sign * rate);
-		}
-	}
-
-	void OnDrawGizmos()
-	{
-		if (target != null)
-		{
-			// draw line to target
-			Gizmos.color = Color.yellow;
-			Gizmos.DrawLine (this.transform.position, target.transform.position);
-
-			// draw ray in facing direction
-			Gizmos.color = Color.red;
-			Gizmos.DrawRay (this.transform.position, this.transform.up);
-		}
-
-		// draw tracking radius
-		Gizmos.color = Color.yellow;
-		float theta = 0;
-		float x = trackingRadius * Mathf.Cos(theta);
-		float y = trackingRadius * Mathf.Sin(theta);
-		Vector3 pos = transform.position + new Vector3(x, y, 0);
-		Vector3 newPos = pos;
-		Vector3 lastPos = pos;
-		for(theta = 0.1f; theta < Mathf.PI * 2; theta += 0.1f)
-		{
-			x = trackingRadius * Mathf.Cos(theta);
-			y = trackingRadius * Mathf.Sin(theta);
-			newPos = transform.position + new Vector3(x, y, 0);
-			Gizmos.DrawLine(pos, newPos);
-			pos = newPos;
-		}
-		Gizmos.DrawLine(pos,lastPos);
-	}
+    public enum targetTypes
+    {
+        friendly,
+        enemy,
+        both
+    }
+
+    static List<GameObject> targets;
+
+    public float degreesPerSecond = 180;
+    public float trackingRadius = 2;
+    public targetTypes targetType = targetTypes.enemy;
+
+    GameObject target = null;
+
+    Team team = null;
+
+    // Use this for initialization
+    void Start ()
+    {
+        // set once and is static. All enemies exist on spawn so we don't want to call this slow function repeatedly
+        if (targets == null) {
+            targets = Object.FindObjectsOfType<Team> ().Select (team => team.gameObject).ToList();
+        }
+        UpdateTarget ();
+    }
+
+    void UpdateTarget()
+    {
+        targets.RemoveAll (target => target == null);
+
+        // I love me some lambdas
+        // Get objects within range
+        List<GameObject> trackableTargets = targets.Where((target) => (target.transform.position - this.transform.position).sqrMagnitude < trackingRadius * trackingRadius).ToList();
+
+        // Filter targets by target type, ignore if no team was set
+        if (this.team)
+        {
+            switch ((int)targetType)
+            {
+            case (int)targetTypes.friendly:
+                trackableTargets = trackableTargets.Where (target =>
+                {
+                    Team team = target.GetComponent<Team> ();
+                    Flag flag = target.GetComponent<Flag>();
+                    return team && team.IsFriendly(this.team) && flag.isTrackable;
```

Comment:
Same thing, check for existence of flag class `... && flag && flag.isTrackable)`